components/src/components/elements/payment-method/PaymentMethod.tsx

```diff
@@ -65,7 +129,162 @@ const PaymentMethodElement = ({
 
   const isLightBackground = useIsLightBackground();
 
-  const sizeFactor = size === "lg" ? 2 : size === "md" ? 1 : 0.5;
+  const sizeFactor = size === "lg" ? 1.6 : size === "md" ? 1 : 0.5;
+
+  const accountLast4 = paymentMethod?.accountLast4;
+  const accountName = paymentMethod?.accountName;
+  const bankName = paymentMethod?.bankName;
+  const billingEmail = paymentMethod?.billingEmail;
+  const cardBrand = paymentMethod?.cardBrand;
+  const cardLast4 = paymentMethod?.cardLast4;
+  const paymentMethodType = paymentMethod?.paymentMethodType;
+
+  const isCard = paymentMethodType === "card";
+  const isBank = paymentMethodType === "us_bank_account";
+  const isAmazonPay = paymentMethodType === "amazon_pay";
+  const isApplePay = paymentMethodType === "apple_pay"; // nothing in stripe docs...
+  const isCashApp = paymentMethodType === "cashtag";
+  const isPayPal = paymentMethodType === "paypal";
+  const isLink = paymentMethodType === "link";
+
+  const getIconName = (value: string | undefined) => {
+    switch (value) {
+      case "visa":
+      case "mastercard":
+      case "amex":
+        return value;
+      default:
+        return "credit";
+    }
+  };
+
+  const getIconStyles = (size: "sm" | "md" | "lg") => {
+    const styles = {
+      lg: { fontSize: 28, marginLeft: -2, marginRight: 8 },
+      md: { fontSize: 25, marginLeft: 0, marginRight: 7, marginTop: -1 },
+      sm: { fontSize: 24, marginLeft: 0, marginRight: 4 },
+    };
+    return styles[size] || styles.sm;
+  };
+
+  const renderPaymentMethodElement = () => {
+    switch (true) {
```

Comment:
rather than `switch (true)` here, why not just switch on `paymentMethod?.paymentMethodType`?